test/unit/property-list.test.js

```diff
@@ -720,6 +720,162 @@ describe('PropertyList', function () {
+        });
+
+        it('should correctly bind the context to the iterator if one is provided', function () {
+            var parent = new PropertyList(FakeType, {}, [{ key: 'level', value: 'parent' }]),
+                pList = new PropertyList(FakeType, parent, [{ key: 'level', value: 'child' }]),
+                chain = [];
+
+            pList.eachParent(function (parent) {
+                parent.members && chain.push(parent.members[0][this.attr]);
+            }, {
+                attr: 'value'
+            });
+
+            expect(chain).to.eql(['parent']);
+        });
+
+        it('should use the default context of the caller instance if none is provided', function () {
+            var parent = new PropertyList(FakeType, {}, [{ key: 'level', value: 'parent' }]),
+                pList = new PropertyList(FakeType, parent, [{ key: 'level', value: 'child' }]),
+                chain = [];
+
+            pList.eachParent(function (parent) {
+                expect(this.toJSON()).to.eql([{ key: 'level', value: 'child' }]);
+                parent.members && chain.push(parent.members[0].value);
+            });
+
+            expect(chain).to.eql(['parent']);
+        });
+    });
+
+    describe('.filter', function () {
+        var FakeType = function (options) {
+            this.key = options.key;
+            this.value = options.value;
+        };
+
+        it('should correctly bind the iterator to the provided context', function () {
+            var pList = new PropertyList(FakeType, {}, [
+                { key: 'foo', value: 'bar' },
+                { key: 'alpha', value: 'bar' },
+                { key: 'gamma', value: 'baz' }
+            ]);
+
+            expect(pList.filter(function (property) {
+                return property.value === this.value;
+            }, { value: 'bar' })).to.eql([
+                { key: 'foo', value: 'bar' },
+                { key: 'alpha', value: 'bar' }
+            ]);
+        });
+    });
+
+    describe('.find', function () {
+        var FakeType = function (options) {
+            this.key = options.key;
+            this.value = options.value;
+        };
+
+        it('should correctly bind the iterator to the provided context', function () {
+            var pList = new PropertyList(FakeType, {}, [
+                { key: 'foo', value: 'bar' },
+                { key: 'alpha', value: 'bar' },
+                { key: 'gamma', value: 'baz' }
+            ]);
+
+            expect(pList.find(function (property) {
+                return property.key === this.key;
+            }, { key: 'alpha' })).to.eql(
+                { key: 'alpha', value: 'bar' }
+            );
+        });
+    });
+
+    describe('.insert', function () {
+        it('should bail out for non-object arguments', function () {
+            var pList = new PropertyList();
+
+            pList.insert();
+            expect(pList.members).to.eql([]);
```

Comment:
That would also pass for an empty object, the test here is strictly for empty arrays.